jupyterhub/templates/scheduling/_scheduling-helpers.tpl

```diff
@@ -0,0 +1,47 @@
+{{- /*
+  jupyterhub.prepareScope
+    Requires the .podKind field.
+
+    prepareScope sets fields on the scope to be used by the
+    jupyterhub.tolerations and jupyterhub.affinity helm template functions
+    below.
+*/}}
+{{- define "jupyterhub.prepareScope" -}}
+{{- /* Update the copied scope */}}
+{{- $dummy := set . "component" (include "jupyterhub.componentLabel" .) }}
+
+{{- /* Fetch relevant information */}}
+{{- $dummy := set . "tolerationsList" (include "jupyterhub.tolerationsList" .) }}
+{{- $dummy := set . "tolerations" (include "jupyterhub.tolerations" .) }}
+{{- $dummy := set . "hasTolerations" .tolerations }}
```

Comment:
Let's try to do this without prepareScope. I think helpers that setup the state for other helpers, rather than being self-contained is something we should try to avoid.

Reply:
Will attempt to address concerns at a single discussion entry, not in a review comment.